.github/workflows/build-and-preview-site.yml

```diff
@@ -19,8 +19,10 @@ jobs:
 
       - name: Deploy and Preview 🚀
         uses: JamesIves/github-pages-deploy-action@4.0.0
+        env:
+          PR_NUMBER: ${{ github.event.number }}
         with:
-          repository-name: "layer5labs/layer5"
+          repository-name: "layer5labs/layer5/{PR_NUMBER}"
```

Comment:
That's an interesting way...🤔

Reply:
Though, it has one drawback and that is what happens in case of multiple PR?

Reply:
I tossed out this suggestion a couple times amidst our conversations and am curious to know if we can be successful. It should be the case that we can. Here's the thought written down - https://github.com/layer5io/layer5/issues/1578.

Initial build works - https://layer5labs.github.io/layer5/

The Gatsby config will need to be tweaked.

Reply:
Multiple PRs = multiple deploys of the site in the preview repo, separated into different root folders (root according to that PR). There is some work to do to either dynamically update the Gatsby config during execution of this workflow.

Reply:
That I understand, but the github preview will be live using any one of all the root folders. Which means, the last PR will be hosted or the latest deployment.
And, what if two PRs are pushed together?

Reply:
There's a solution in here somewhere. The PR # can be dynamically retrieved. A new folder with PR # as its name (guaranteed to be non-colliding) can be dynamically created (and deleted when PR is merged or closed).

Use of `--path-prefix` or other config might be necessary. The collection of built files and folders under `public` just need an http server to be served from, so stuffing copies of the site into individual folders and dropping a preview link into a comment on the PR at the end of the build-and-preview workflow should give us the DX we're looking for without restriction on minutes and $.

```
module.exports = {
  pathPrefix: `/yourpathprefix`,
}
```